### PR TITLE
fix: builds on various archs/platforms re: PR 4331

### PR DESCRIPTION
**Pull request by wileyj**
ref:  https://github.com/stacks-network/stacks-core/pull/4331

these are the equivalent changes for `master` and `develop`. 

fix building & cross compilation, including stacks-signer, tested building for the following:

- linux-glibc-x86-64
- linux-glibc-arm64
- linux-glibc-armv7
- linux-musl-x86_64
- linux-musl-arm64
- linux-musl-armv7
- windows-x86_64
- macos-x86_64
- macos-arm64

*note*: target branch here is for develop since there is a workaround in place already in `master`. this change won't affect that workaround the next time develop is merged to master

cc: @zone117x @hugocaillard

### CODE REVIEW

**Review by zone117x — APPROVED** [review]

File: stacks-signer/Cargo.toml [hunk 1]
```diff
@@ -27,9 +27,9 @@ clap = { version = "4.1.1", features = ["derive", "env"] }
 hashbrown = "0.14"
 libsigner = { path = "../libsigner" }
 libstackerdb = { path = "../libstackerdb" }
-p256k1 = "5.5"
+p256k1 = { version = "5.5", default-features = false }
 rand_core = "0.6"
-reqwest = { version = "0.11.22", features = ["blocking", "json"] }
+reqwest = { version = "0.11.22", default-features = false, features = ["blocking", "json", "rustls-tls"] }
```
Comment (jcnelson) [inline stacks-signer/Cargo.toml:32]: Is there a way to simply build without TLS?  If not, then this is fine.

**Review by jcnelson — APPROVED** [review]

**Comment by wileyj** [comment]
following discussion in #4331 , converting this to draft. 
there may be another option vs modifying the cargo.toml's here - eg using a native arm runner to build the binaries.

**Comment by wileyj** [comment]
@xoloki can you confirm this is OK to do, or should be continue looking for build alternatives where we're not cross-compiling?

**Review by hstove — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -11,6 +11,12 @@ members = [
     "stacks-signer",
     "testnet/stacks-node"]
 
+# Dependencies we want to keep the same between workspace members
+[workspace.dependencies]  
+wsts = { version = "7.0", default-features = false }
```
Comment (xoloki) [inline Cargo.toml:16]: This is the correct way to include `wsts` to avoid including `bindgen` and friends.

File: stacks-signer/Cargo.toml [hunk 3]
```diff
@@ -42,7 +42,7 @@ thiserror = "1.0"
 toml = "0.5.6"
 tracing = "0.1.37"
 tracing-subscriber = { version = "0.3.17", features = ["env-filter"] }
-wsts = "4.0.0"
+wsts = { version = "4.0.0", default-features = false }
```
Comment (xoloki) [inline stacks-signer/Cargo.toml:45]: Is there a reason why we have such an old version of `wsts` here in the `stacks-signer` `Cargo.toml`?  And why we have different versions here and in the root `Cargo.toml`?

**Review by xoloki — APPROVED** [review]
The general approach of using `default-features = false` is correct.

I'm a little concerned that we are using different versions of `wsts` in the root and `stacks-signer` `Cargo.toml` files, but that was present before this PR so not a blocker.

File: stacks-signer/Cargo.toml (hunk 3 above)
Comment (wileyj) [reply stacks-signer/Cargo.toml:45]: Can't say for certain, but best to leave as-is since this branch has been genesis-sync tested as working.

**Comment by wileyj** [comment]
> The general approach of using `default-features = false` is correct.
> 
> I'm a little concerned that we are using different versions of `wsts` in the root and `stacks-signer` `Cargo.toml` files, but that was present before this PR so not a blocker.

any concerns with the similar PR https://github.com/stacks-network/stacks-core/pull/4331 ?

**Comment by xoloki** [comment]
> > The general approach of using `default-features = false` is correct.
> > I'm a little concerned that we are using different versions of `wsts` in the root and `stacks-signer` `Cargo.toml` files, but that was present before this PR so not a blocker.
> 
> any concerns with the similar PR #4331 ?

#4331 is for `next` which uses a workspace dependency for `wsts` so no concerns there.

**Pull request merged**